llvmcpy/llvm.py

```diff
@@ -257,6 +258,28 @@ def call():
         if failure != 0:
             raise LLVMException({})
         return result""".format(result_type, call(), error_message)
+    elif (return_type.kind == "pointer"
+          and return_type.item.kind == "primitive"
+          and return_type.item.cname == "char"
+          and last_arg and last_arg.kind == "pointer"
+          and last_arg.item.kind == "primitive"
+          and last_arg.item.cname in ("unsigned int", "unsigned long")):
+        # Special case: the function returns a string and
+        # the last argument is a pointer to an unsigned int/long:
+        # This often represents returning a string (that may contain
+        # NUL characters) with a length
+        function_arguments[-1] = None
+        arguments[-1] = "lengthbuf"
```

Comment:
It's a buffer to hold the length of the returned string, so `length_buffer` might be more clear.